tests/test_recipes/test_core_recipe.py

```diff
@@ -0,0 +1,231 @@
ExZWXnLli3V1dUYhuGuqhBYHQ7xODt8+HA6nY5/nTp1qqqq6vbt2+EFT0lJyYCAgBcvXhCjAd4KA9/qbACpqalRVVXt7Ox8/fo1hmHl5eWkxxm+OlAkiC3kJbePQ/TBINbFpz0Yj2G3tbXFn17BMCwxMdHe3h6vBR92Go3W3d19/PhxBwcH6Dr74MEDzobxEdrntQkuXLiAYdjIkSPt7e1hk4YNG/b48WMMwz59+tTa2gp7N3z4cOjVALPjd8DV+AIfOviBl6g/18aT5glpq+GRSSMwfPjwQ4cOvXr1at++fXPmzCHFDA8Ph4OwefPmadOm1dTUKCsrt7a21tTUwAJEE1ecwYMH29vb79mzx8TEBC5ZuXIl9LbFMIxOp3d0dEybNk1NTS0tLQ3fpjBUbGysgYHBxIkTDxw4AF0JIWPHjl2+fLmOjg50u/2qjwfJnYDPTCAiIyNz584duGJmZibJKYLrTMC4OcYQB19aWjo9Pb2pqQnDMNyeAgBAqgtfFw/F1fwELwMjkCITIblh4COAYVhra+vs2bM9PT1jYmJ8fX35j+R/kX6b+ebPnx8eHh4fH5+YmPj58+fy8nIPD4/4+PikpKS0tDRnZ+fi4mKo0//s2TOo03/y5Eno3s6JgoLC8ePHr1y5EhISkpeXB93Mvb297927FxAQ8OjRo6ioKCaTmZyc3NjYCBX0IyMjoTv52bNnIyMjg4KCMAyDjl+3b9++d++eqalpd3f38+fPKysrKRRKeHi4jY0N9IIpKiqqqKjg5eouIiICnQ2ePHkCnQ3Wrl0bGhp69uxZOp0O//hDZ4O2tjYWi/XXX3+FhoZu2rRp4sSJ58+f19HRMTIySk1NZbFYnz59evz4MZ1Of/XqVU1NTWRkJJvNJg3UkydPoOI+i8VKT0//8uVLTU1NVlbW69evocXEx48fGxoaHB0daTQaSW4fB7bn6dOnzc3NL1++TE5Obmtr+/TpE7E9nA9/cx32kydPSklJubi4PH78+M8//1y1apWFhQWGYZ2dnTExMXV1ddDCbe/evVQqFRq5tbS0TJo06fjx4+Xl5aQq7O3t6XT6vXv3/P39/fz8YKsyMzPLysqgNQdnX16+fHny5MmoqKgbN25AG3F3d3dLS8s1a9akpaUBADIzMzMzM3/cNZ6r8QX0HsFtLgYPHgxF/S0tLel0ellZGWlWEwPOnj0bOmAUFhYWFRXV1dXhW621tRU2m0KhkEagrq7O1dU1KCjo8OHDnJ6O8vLyO3fuDAoKcnd3nzJlioaGhoiISEhIiIODQ1RU1P379+H/OXzXwFe0srLatGmTrKws/Dp58uQ9e/Y4Ojo+fPjw8OHDISEhwsLCqqqqlpaWe/bsgbsYPNULCgqaNGnShAkTli5dGhYWlpGR0dLS8u7du/T09NDQ0KSkJOjqvGvXLpgCuYK7E+BPMpNmQnV1Nb53EFfU09OjUqmqqqp2dnYTJkyQkJAIDg6+cuUKvIFNnAl0Or2oqCglJaWsrKy0tJSUeNrb2/HBl5GROXLkyKxZs/T19Xt7e4uLi6HdB6kukttMe3v74sWLz58/n/k3TU1N2dnZOTk5dXV1OTk5r169EhMTI0YmDQJ0wzh+/Hhubu779+/nz59vbGzs5uYWHR1tYGAAfd77K/1ZvYyrLQAfIXk2mw0f6+D6a3d3t7CwcF1d3ZgxY4gH987OziFDhnCaEvzTkJwNoKI8LvSOOxuMHj26oqKCzWbjQwH/OH+1wXwGihPAW24fQvTBIMG/PbyGncVi1dXV4dfKfhCS0D5/oClBdXW1tLQ03mwmk9nT08NLgP+74W98gfNNov64A8ZXtxokIyMDADB27NiWlpaHDx+eOnWK+Ct0SKDRaF1dXfAFSpyWlhYJCQk+o8pgMIhXOGHzWltbSde9uS78KtDY4VunRx9nApVKxXc9kvVKH2cC5+DDqc55cZhYFyd8zE9weEXGONwwYMCBAwf++gPaL6Y/Z75vhUKhSElJ/RrXm18Dm80WFxevra3t42ERgeDEwcGhs7Nz48aN9fX1srKy0KYVgfhPgzJff+bNmzd5eXmysrLr1q37t9uC+K/CYrHevHlTXV09b968cePG/dvNQSB+AijzIRAIBEKw6LdPuCAQCAQCwRWU+RAIBAIhWKDMh0AgEAjBAmU+BAKBQAgWKPMhEAgEQrBAmQ+BQCAQggXKfAgEAoEQLFDmQyAQCIRggTIfAoFAIAQLlPkQCAQCIVigzIdAIBAIwQJlPgQCgUAIFijzIRAIBEKwQJkPgUAgEIIFynwIBAKBECxQ5kMgEAiEYIEyHwKBQCAEC5T5EAgEAiFYoMyHQCAQCMECZT4EAoFACBYo8yEQCARCsECZD4FAIBCCBcp8CAQCgRAsUOZDIBAIhGCBMh8CgUAgBAuU+RAIBAIhWKDMh0AgEAjBAmU+BAKBQAgWKPMhEAgEQrBAmQ+BQCAQggXKfAgEAoEQLFDmQyAQCIRggTIfAoFAIAQLlPkQCAQCIVigzIdAIBAIwQJlPgQCgUAIFijzIRAIBEKwQJkPgUAgEIIFynwIBAKBECxQ5kMgEAiEYIEyHwKBQCAEC5T5EAgEAiFYoMyHQCAQCMECZT4EAoFACBYo8yEQCARCsECZD4FAIBCCBcp8CAQCgRAsUOZDIBAIhGCBMh8CgUAgBAuU+RAIBAIhWKDMh0AgEAjBAmU+BAKBQAgWKPMhEAgEQrBAmQ+BQCAQggXKfAgEAoEQLFDmQyAQCIRggTIfAoFAIAQLlPkQCAQCIVigzIdAIBAIwQJlPgQCgUAIFijzIRAIBEKwQJkPgUAgEIIFynwIBAKBECxQ5kMgEAiEYIEyHwKBQCAEC5T5EAgEAiFYoMyHQCAQCMECZT4EAoFACBYo8yEQCARCsECZD4FAIBCCBcp8CAQCgRAsUOZDIBAIhGCBMh8CgUAgBAuU+RAIBAIhWKDMh0AgEAjB4v8BmGoNiKO4Gz0AAAAASUVORK5CYII='
```

Comment:
could possibly pull the base64 page image into a separate resource so test is easier to read...for the other expected outputs, I think it enhances understanding to see them